Adding dbt-watsonx-presto setup and config files (#6736)
fishtown-analytics/docs.getdbt.com: website/docs/docs/core/connect-data-platform/watsonx-presto-setup.md

---
title: "IBM watsonx.data Presto setup"
description: "Read this guide to learn about the IBM watsonx.data Presto setup in dbt."
id: "watsonx-presto-setup"
meta:
  maintained_by: IBM
  authors: Karnati Naga Vivek, Hariharan Ashokan, Biju Palliyath, Gopikrishnan Varadarajulu, Rohan Pednekar
  github_repo: 'IBM/dbt-watsonx-presto'
  pypi_package: 'dbt-watsonx-presto'
  min_core_version: v1.8.0
  cloud_support: 'Not Supported'
  min_supported_version: 'n/a'
  slack_channel_name: 
  slack_channel_link: 
  platform_name: IBM watsonx.data
  config_page: /reference/resource-configs/watsonx-presto-config
---

The dbt-watsonx-presto adapter allows you to use dbt to transform and manage data on IBM watsonx.data Presto(Java), leveraging its distributed SQL query engine capabilities. Before proceeding, ensure you have the following:
<ul>

>   <li>An active IBM watsonx.data Presto(Java) engine with connection details (host, port, catalog, schema) in SaaS/software.</li>

Reviewer: ```suggestion
  <li>An active IBM watsonx.data Presto(Java) engine with connection details (host, port, catalog, schema) in SaaS/Software.</li>
```

  <li>Authentication credentials: Username and password/apikey.</li>
  <li>For watsonx.data instances, SSL verification is required for secure connections. If the instance host uses HTTPS, there is no need to specify the SSL certificate parameter. However, if the instance host uses an unsecured HTTP connection, ensure you provide the path to the SSL certificate file.</li>
</ul>
Refer to [Configuring dbt-watsonx-presto](https://www.ibm.com/docs/en/watsonx/watsonxdata/2.1.x?topic=presto-configuration-setting-up-your-profile) for guidance on obtaining and organizing these details.

import SetUpPages from '/snippets/_setup-pages-intro.md';

<SetUpPages meta={frontMatter.meta}/>

## Connecting to IBM watsonx.data presto

> To connect dbt with watsonx.data Presto(java), you need to configure a profile in your `profiles.yml` file located in the `.dbt/` directory of your home folder. The following is an example configuration for connecting to IBM watsonx.data SaaS and software instances:

Reviewer: ```suggestion
To connect dbt with watsonx.data Presto(java), you need to configure a profile in your `profiles.yml` file located in the `.dbt/` directory of your home folder. The following is an example configuration for connecting to IBM watsonx.data SaaS and Software instances:
```

<File name='~/.dbt/profiles.yml'>

```yaml
my_project:
  outputs:
    software:
      type: presto
      method: BasicAuth
      user: [user]
      password: [password]
      host: [hostname]
      database: [catalog name]
      schema: [your dbt schema]
      port: [port number]
      threads: [1 or more]
      ssl_verify: path/to/certificate

    saas:
      type: presto
      method: BasicAuth
      user: [user]
      password: [api_key]
      host: [hostname]
      database: [catalog name]
      schema: [your dbt schema]
      port: [port number]
      threads: [1 or more]

  target: software

```

</File>

## Host parameters

> The following profile fields are required to configure watsonx.data Presto(java) connections. For IBM watsonx.data SaaS or software instances, you can get the `hostname` and `port` details by clicking **View connect details** on the Presto(java) engine details page.

Reviewer: ```suggestion
The following profile fields are required to configure watsonx.data Presto(java) connections. For IBM watsonx.data SaaS or Software instances, you can get the `hostname` and `port` details by clicking **View connect details** on the Presto(java) engine details page.
```

| Option    | Required/Optional | Description | Example  |
| --------- | ------- | ------- | ----------- |

> | `method`  | Required | Specifies the authentication method for secure connections. Use `BasicAuth` when connecting to IBM watsonx.data SaaS or software instances. | `BasicAuth` |

Reviewer: ```suggestion
| `method`  | Required | Specifies the authentication method for secure connections. Use `BasicAuth` when connecting to IBM watsonx.data SaaS or Software instances. | `BasicAuth` |
```

---
title: "IBM watsonx.data Presto setup"
description: "Read this guide to learn about the IBM watsonx.data Presto setup in dbt."

> id: "watsonx-presto setup"

Reviewer 2: ```suggestion
id: "watsonx-presto-setup"
```

meta:
  maintained_by: IBM
  authors: Karnati Naga Vivek, Hariharan Ashokan, Biju Palliyath, Gopikrishnan Varadarajulu, Rohan Pednekar
  github_repo: 'IBM/dbt-watsonx-presto'
  pypi_package: 'dbt-watsonx-presto'
  min_core_version: v1.8.0
  cloud_support: 'Not Supported'
  min_supported_version: 'n/a'
  slack_channel_name: 
  slack_channel_link: 
  platform_name: IBM watsonx.data
  config_page: /reference/resource-configs/watsonx-presto-config
---

The dbt-watsonx-presto adapter allows you to use dbt to transform and manage data on IBM watsonx.data Presto(Java), leveraging its distributed SQL query engine capabilities. Before proceeding, ensure you have the following:
<ul>

>   <li>An active IBM watsonx.data Presto(Java) Engine with connection details (host, port, catalog, schema) in SaaS/Software.</li>

Reviewer: ```suggestion
  <li>An active IBM watsonx.data Presto(Java) engine with connection details (host, port, catalog, schema) in SaaS/software.</li>
```

Author: @nataliefiann In doc pages the term Software means IBM watsonx.data on-premises software. So i think we need to mention that word as **Software** only

Reviewer 3: @KNagaVivek just to be clear - Software is like a product name here?

Author: yes @amychen1776

>   <li>Authentication Credentials: Username and password/apikey.</li>

Reviewer: ```suggestion
  <li>Authentication credentials: Username and password/apikey.</li>
```

  <li>For watsonx.data instances, SSL verification is required for secure connections. If the instance host uses HTTPS, there is no need to specify the SSL certificate parameter. However, if the instance host uses an unsecured HTTP connection, ensure you provide the path to the SSL certificate file.</li>
</ul>

> Refer to the Configuring dbt-watsonx-presto section for guidance on obtaining and organizing these details.

Reviewer: ```suggestion
Refer to [Configuring dbt-watsonx-presto](https://www.ibm.com/docs/en/watsonx/watsonxdata/2.1.x?topic=presto-configuration-setting-up-your-profile) for guidance on obtaining and organizing these details.
```

<Snippet path="warehouse-setups-cloud-callout" />

import SetUpPages from '/snippets/_setup-pages-intro.md';

<SetUpPages meta={frontMatter.meta}/>

## Connecting to IBM watsonx.data presto

> To connect dbt with watsonx.data Presto(java), you need to configure a profile in your `profiles.yml` file located in the `.dbt/` directory of your home folder. The following is an example configuration for connecting to IBM watsonx.data SaaS and Software instances:

Reviewer: ```suggestion
To connect dbt with watsonx.data Presto(java), you need to configure a profile in your `profiles.yml` file located in the `.dbt/` directory of your home folder. The following is an example configuration for connecting to IBM watsonx.data SaaS and software instances:
```

<File name='~/.dbt/profiles.yml'>

```yaml
my_project:
  outputs:
    software:
      type: presto
      method: BasicAuth
      user: [user]
      password: [password]
      host: [hostname]
      database: [catalog name]
      schema: [your dbt schema]
      port: [port number]
      threads: [1 or more]
      ssl_verify: path/to/certificate

    saas:
      type: presto
      method: BasicAuth
      user: [user]
      password: [api_key]
      host: [hostname]
      database: [catalog name]
      schema: [your dbt schema]
      port: [port number]
      threads: [1 or more]

  target: software

```

</File>

## Host parameters

> The following profile fields are required to configure watsonx.data Presto(java) connections. For IBM watsonx.data SaaS or Software instances, you can get the `hostname` and `port` details by clicking **View connect details** on the Presto(java) engine details page.

Reviewer: ```suggestion
The following profile fields are required to configure watsonx.data Presto(java) connections. For IBM watsonx.data SaaS or software instances, you can get the `hostname` and `port` details by clicking **View connect details** on the Presto(java) engine details page.
```

> The following profile fields are required for configuring watsonx.data Presto(java) connections. Currently, it supports only the `BasicAuth` authentication method. For IBM watsonx.data SaaS or Software instances, You can get the hostname and port details by clicking View connect details inside the Presto(java) engine details page.

Reviewer 3: ```suggestion
The following profile fields are required for configuring watsonx.data Presto(java) connections. Currently, it supports only the `BasicAuth` authentication method. For IBM watsonx.data SaaS or Software instances, you can get the hostname and port details by clicking View connect details inside the Presto(java) engine details page.
```

Reviewer 3: This paragraph is also a tad confusing to me - does the adapter support only `BasicAuth` for authentication? Also I assume host and port details are required fields regardless ?

Author: Yes, host and port details are required fields for the connection. The adapter uses `none` as the default value for the authentication method, which is helpful for connecting to OSS Presto instances with local connections that don’t require authentication.

Reviewer 3: Oh I see - given that the focus of this page is just on IBM Watson, to keep it clean to understand I recommend removing OSS Presto mentions and just how to set this up for IBM Watson

Author: @amychen1776  Updated the docs by removing OSS Presto mentions. can you pls review ?

| Option    | Required/Optional | Description | Example  |
| --------- | ------- | ------- | ----------- |

> | `method`  | Required (default value is none) | Authentication method for Presto | `None` or `BasicAuth` |

Reviewer 3: Would the user ever want to put in `None`?

Author: Using `None` for authentication is practical for local or unauthenticated OSS Presto setups. For secure environments, users would typically use `BasicAuth` method.

Reviewer 3: same as what I mentioned above - let's remove the loss presto setups since that's not the focus of the doc

Author: Updated the docs by removing OSS Presto mentions.

|   `user`  | Required | Username or email for authentication. | `user` |

> | `password`| Required (if `method` is `BasicAuth`) | Password or API key for authentication | `password` |

Reviewer 3: If the adapter only supports basicauth - I would just put this as required

Author: Updated accordingly.

|   `host`  | Required | Hostname for connecting to Presto. | `127.0.0.1` |

> | `database`| Required | The catalog name in your presto instance. | `Analytics` |

Reviewer: ```suggestion
| `database`| Required | The catalog name in your Presto instance. | `Analytics` |
```

> |  `schema` | Required | The schema name within your presto instance catalog. | `my_schema`  |

Reviewer: ```suggestion
|  `schema` | Required | The schema name within your Presto instance catalog. | `my_schema`  |
```

> |  `schema` | Required | The schema name within your presto instance catalog. | `my_schema`  |

Reviewer: ```suggestion
|  `schema` | Required | The schema name within your Presto instance catalog. | `my_schema`  |
```
